yaptide/celery/tasks.py

```diff
@@ -134,29 +136,32 @@ def read_file(stats: SimulationStats, filepath: Path, task_id: int):
 
 
 @celery_app.task(bind=True)
-def run_simulation(self, json_data: dict):
+def run_simulation(self, payload_dict: dict):
     """Simulation runner"""
     # create temporary directory
     with tempfile.TemporaryDirectory() as tmp_dir_path:
+        runner_obj = SHRunner(jobs=payload_dict.get("ntasks"),
+                              keep_workspace_after_run=True,
+                              output_directory=tmp_dir_path)
+
+        # since it is not obligatory for UI to provide ntasks parameters it is set here for sure
```

Comment:
we can wait as well for UI modification and assume that valid payload_dict should contain `ntasks`
otherwise I'd assume that is `ntasks` is not set in the payload it should be defaulted to 1 task here